src/Elf.cpp

```diff
@@ -139,56 +139,49 @@ struct Rel {
     typedef typename T::addr_t addr_t;
     typedef typename T::addr_off_t addr_off_t;
 
-    addr_t r_offset;
-    addr_t r_info;
+    const addr_t r_offset;
+    const addr_t r_info;
 
     uint32_t r_type() const {
-        if (sizeof(addr_t) == 8) {
+        if constexpr (sizeof(addr_t) == 8) {
```

Comment:
I'm not sure what these are here for. I don't think the affect code generation and putting the assertion on a sizeof check seems more noise than useful to me.